docs/checks.rst

```diff
@@ -52,3 +52,23 @@ PostgreSQL table once.
 
 Read-write mappings in Heroku Connect need to provide an upsert field to
 identify the record.
+
+``heroku_connect.E008``
+~~~~~~~~~~~~~~~~~~~~~~~
+
+A related field (:class:`ForeignKey<django.db.models.ForeignKey>` or
+:class:`ManyToManyField<django.db.models.ManyToManyField>`) pointing to
+a Heroku Connect model may not use database-constraints.
+
+In some cases, such as version updates, Heroku Connect recreates the tables,
+which will break if there are database constraints pointing to them.
```

Comment:
I think, it would be good to mention official HC doc about this somewhere...
https://devcenter.heroku.com/articles/heroku-connect-logs-errors#error-during-sync-psycopg2-integrityerror-update-or-delete-on-table-mycustomobject__c-violates-foreign-key-constraint